### PR TITLE
Issue 88 Add new Tenants APIs

### PR DESCRIPTION
**Pull request by tungleduyxyz**
Related issue: https://github.com/killbill/killbill-client-ruby/issues/88
Updates:
 - Tenant push notifications
 - Tenant System properties configuration
 - Tenant Payment States Machine configuration

### CODE REVIEW

File: lib/killbill_client/models/tenant.rb [hunk 1]
```diff
@@ -126,6 +129,42 @@ def create(use_global_default=true, user = nil, reason = nil, comment = nil, opt
         options[:api_secret] = @api_secret
         created_tenant.refresh(options)
       end
+
+      def register_push_notification(cb, options = {})
+        post "#{KILLBILL_API_TENANTS_NOTIFICATION}", {}, {:cb => cb}, options
+      end
+
+      def retrieve_push_notification(options = {})
+        get "#{KILLBILL_API_TENANTS_NOTIFICATION}", {}, options
```
Comment (reshmabidikar) [inline lib/killbill_client/models/tenant.rb:138]: Other tenant methods have a call to `require_multi_tenant_options`. See [this line](https://github.com/killbill/killbill-client-ruby/blob/7ce8d28c5bf608cccd48d5b9d4215f77e81282c7/lib/killbill_client/models/tenant.rb#L61). Is this required here and for other methods?

**Review by reshmabidikar — COMMENTED** [review]
I've added some comments. Also, I would suggest that you also execute the ruby code and confirm that it works as expected if not already done.

**Comment by tungleduyxyz** [comment]
Thanks @reshmabidikar . I have updated the code, please help me review it again.

**Review by reshmabidikar — APPROVED** [review]
Looks good!

**Pull request merged**